### PR TITLE
fix(eap): Adjust weighted functions to account for missing values

### PR DESCRIPTION
**Pull request by Zylphrex**
Similar to #79141, this handles missing values for the weighted functions on eap.

### CODE REVIEW

File: src/sentry/search/events/datasets/spans_indexed.py [hunk 1]
```diff
@@ -769,10 +736,18 @@ def function_converter(self) -> dict[str, SnQLFunction]:
...
                     "count_weighted",
-                    optional_args=[NullColumn("column")],
+                    optional_args=[
+                        with_default("span.duration", NumericColumn("column", spans=True)),
+                    ],
                     snql_aggregate=self._resolve_count_weighted,
                     default_result_type="integer",
                 ),
+                SnQLFunction(
+                    "count_unique_weighted",
```
Comment (shruthilayaj) [inline src/sentry/search/events/datasets/spans_indexed.py:746]: why are we adding this? 👀 we can't really weight count_unique right

Comment (shruthilayaj) [reply src/sentry/search/events/datasets/spans_indexed.py:746]: I see we have them for min/max below as well

**Review by shruthilayaj — APPROVED** [review]
a couple small questions, but LGTM

File: src/sentry/search/events/datasets/spans_indexed.py [hunk 2]
```diff
@@ -618,19 +588,16 @@ def resolve_aggregate_if(
...
 
-            unprefixed = Function("mapContains", [attr_col, attr_name])
-            prefixed = Function("mapContains", [attr_col, f"sentry.{attr_name}"])
-
             return Function(
                 function,
                 [
                     args["column"],
-                    Function("or", [unprefixed, prefixed]),
+                    Function("mapContains", [attr_col, attr_name]),
```
[the remark this answers is not in the document]
Comment (Zylphrex) [reply src/sentry/search/events/datasets/spans_indexed.py:600]: The cluster is going to be reset soon anyways so no reason to keep this long winded hack. I'll get around to updating the actual definitions to directly use the `sentry.` prefix as needed soon.

File: src/sentry/search/events/datasets/spans_indexed.py (hunk 1 above)
Comment (Zylphrex) [reply src/sentry/search/events/datasets/spans_indexed.py:746]: It's not actually weighted but these exist for the sake of completeness so the frontend doesn't need to think about whether or not a function can be weighted or not.

**Pull request merged**